cardano-constitution/src/Cardano/Constitution/Validator/Data/Common.hs

```diff
@@ -0,0 +1,91 @@
+-- editorconfig-checker-disable-file
+{-# LANGUAGE LambdaCase        #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE Strict            #-}
+{-# LANGUAGE TypeApplications  #-}
+{-# LANGUAGE ViewPatterns      #-}
+module Cardano.Constitution.Validator.Data.Common
+    ( withChangedParams
+    , ChangedParams
+    , ConstitutionValidator
+    , validateParamValue
+    ) where
+
+import Cardano.Constitution.Config
+import Data.Coerce
+import PlutusLedgerApi.Data.V3
+import PlutusTx.Builtins qualified as B
+import PlutusTx.Builtins.Internal qualified as BI
+import PlutusTx.NonCanonicalRational as NCRatio
+import PlutusTx.Prelude as Tx hiding (toList)
+
+type ConstitutionValidator = ScriptContext -- ^ Deep inside is the changed-parameters proposal
+                           -> BuiltinUnit -- ^ No-error means the proposal conforms to the constitution
+
+-- OPTIMIZE: operate on BuiltinList<BuiltinPair> directly, needs major refactoring of sorted&unsorted Validators
+type ChangedParams = [(BuiltinData, BuiltinData)]
+
+{- HLINT ignore "Redundant lambda" -} -- I like to see until where it supposed to be first applied.
+{- HLINT ignore "Collapse lambdas" -} -- I like to see and comment on each arg
+withChangedParams :: (ChangedParams -> Bool) -> ConstitutionValidator
+withChangedParams fun (scriptContextToValidGovAction -> validGovAction) =
+    case validGovAction of
+        Just cparams ->  if fun cparams
+            then BI.unitval
+            else traceError "ChangedParams failed to validate"
+        Nothing -> BI.unitval -- this is a treasury withdrawal, we just accept it
+{-# INLINABLE withChangedParams #-}
+
+validateParamValue :: ParamValue -> BuiltinData -> Bool
```

Comment:
I think validateParamValue can be imported from NonData.Common since it seems to be the same